sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -1056,7 +1056,10 @@ private Block BuildConditionalExpression(ConditionalExpressionSyntax conditional
         private Block BuildCoalesceExpression(BinaryExpressionSyntax expression, Block currentBlock)
         {
             var rightBlock = BuildExpression(expression.Right, CreateBlock(currentBlock));
-            return BuildExpression(expression.Left, CreateBinaryBranchBlock(expression, rightBlock, currentBlock));
+            var leftCurrentBlock = expression.Parent.Kind() == SyntaxKind.EqualsValueClause ? currentBlock : CreateBlock(currentBlock);

```

Comment:
could you please add a comment why you do this here and not in the SE engine (what we discussed y-day)